### ISSUE TITLE
BEATs extract feature produce nan value

### ISSUE THREAD

**Issue by CrazyGeG** [microsoft/unilm#1672]
**Describe the bug**
I'm using BEATs as audio encoder

The problem arises when using:
BEATs_model = torch.load(beats_checkpoint, map_location=device)  
BEATs_cfg = BEATsConfig(BEATs_model['cfg'])
self.BEATs_model = BEATs(BEATs_cfg)
self.BEATs_model.load_state_dict(BEATs_model['model'])
self.BEATs_model.to(device)  
self.BEATs_model.eval()
for chunk in chunks:
            if chunk.size(1) < chunk_size:  
                # Pad the last chunk
                padding = torch.zeros((1, chunk_size - chunk.size(1)), device = chunk.device)
                chunk = torch.cat((chunk, padding), dim=1)
          for name, module in self.BEATs_model.named_modules():
                def hook(module, input, output):
                    if len(input) > 0 and isinstance(input[0], torch.Tensor):
                        print(f"{module.__class__.__name__} Input Stats: Min: {input[0].min().item()}, Max: {input[0].max().item()}, Mean: {input[0].mean().item()}, Std: {input[0].std().item()}")
                    if isinstance(output, tuple):
                        output = output[0]
                    if torch.isnan(output).any():
                        print(f"NaN detected in layer: {module.__class__.__name__}")
                        raise RuntimeError(f"NaN detected in layer {name}")
            with torch.no_grad():
                            print("Encoder audio:",self.BEATs_model.extract_features(chunk)[0] )

Encoder audio is all nan values and I have tried normalize the chunk, still get nan values. The nan values first detected at layernorm. I tried assigning all 1s to chunk and it worked not producing nan values.

**Comment by paddy0914** [comment]
@CrazyGeG How was it resolved? I encountered the same problem.
